fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1074,11 +1074,19 @@ private static void parseSearchResultParameter(Class<?> resourceType, FHIRSearch
             if (SearchConstants.COUNT.equals(name)) {
                 int pageSize = Integer.parseInt(first);
 
-                // If the user specified a value > max, then use the max.
-                if (pageSize > SearchConstants.MAX_PAGE_SIZE) {
-                    pageSize = SearchConstants.MAX_PAGE_SIZE;
+                if (pageSize < 0) {
+                    throw new IllegalArgumentException("pageSize must be greater than or equal to zero");
+                } else if (pageSize == 0) {
+                    // if _count has the value 0, this shall be treated the same as _summary=count
+                    // https://www.hl7.org/fhir/r4/search.html#count
+                    context.setSummaryParameter(SummaryValueSet.COUNT);
```

Comment:
leaving one comment - makes sense to me - do we want a test for it?

Reply:
will this change the link in the search result? I mean will this append _summary=count there? can mot remember clearly does this impact or not ...

Reply:
> do we want a test for it?
i debated it, but wanted to get this out fast.  now i circled back and added a single e2e test for it

> will this change the link in the search result?
yes, it does affect the self url and I think this is actually a GOOD thing:
`https://localhost:9443/fhir-server/api/v4/Practitioner?_count=10&_id=e98d6843-2a73-46b7-889f-4c097da8f267&_summary=count&_page=1`

Reply:
request _count=0 but response with _count=10&_summary=count seems a little bit weird for me, but not a block issue. I think it's LGTM.